cnd/src/swap_protocols/rfc003/create_swap.rs

```diff
@@ -33,7 +33,7 @@ use genawaiter::{
 ///
 /// It is highly unlikely for Bob to fund the HTLC now, yet the current
 /// implementation is still waiting for that.
-pub async fn create_swap<D, A, AI, BI>(
+pub async fn create_alpha_watcher<D, A, AI, BI>(
```

Comment:
These two functions `create_alpha_watcher` and `create_beta_watcher` are now identical except for a single function call (and alpha/beta in the function comment).  Can we have some abstraction over this code please?  One idea would be to pass the function called (i.e., `watch_foo_ledger`) as an argument, not sure if this works with the turbo fish syntax needed but off the top of my head I don't see why not.

Reply:
Yeah, I discussed this with @thomaseizinger and we would go for a follow up PR it that's OK.

Reply:
The split is possible but I think not trivial because of the `SwapEvent` type still requiring `AL, BL, AA, BA` and its use in `StateStore`.

In any case, the dupliation is an implementation detail and not that critical I would say :)

This is a nice step towards #1953, where we are going to need such functions, albeit they have to be ledger independent for that.

More importantly, this fixes a crucial runtime bug where we shared the instance of the Coroutine between the two watchers and is a blocker for @da-kami's feature. Hence, we opted not to scope creep to much and try to land the actual ticket that is motivating this :)

As always, if you have the time and will, you are happy to send a follow-up PR ;)